### PR TITLE
Narrative pages with animated transitions

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

Builds out the bulk (but not all) of the basic layout UI for the narrative pages. 

Screen Shot 2021-01-07 at 10 49 14 AM

Screen Shot 2021-01-07 at 10 49 38 AM

Functionality you should see: 

- all the content is present on each page
- the sections smoothly "snap" into place as you scroll up and down the page
- up/down chevrons on the left can also be used to page through the sections
- the page progress bar in the left navigation updates as the page scrolls
- the URL updates with the section number as the page scrolls
- if you navigate directly to a URL with a section number (paste it in or just reload the page), the page will scroll to that section once it loads

The animations are somewhat inspired by [this article](https://loremipsum.ueno.co/dear-ueno-how-did-you-make-that-slick-panelsnap-on-your-site-b64160baa7e4), although I didn't really use any of their code directly. We will probably wind up implementing a similar "wipe" transition on this site eventually, but full motion designs are still TBD. I'm also using a different animation library than they did, [React Spring](https://www.react-spring.io/), which has a nice, concise API and seems to be the newest hotness? Seems good so far in this limited context. (And, after a brief scare, has been verified to work in IE 11!)

There is some slightly funky state management within this page to integrate the animations and prevent the router from clobbering the page (the section "routes" are kind of fake since they do not actually map to different components), but I tried to comment voluminously to keep it sensible.

Other work to do on this page (other than the charts, obviously) includes adding a page footer for navigating to the other narratives and adding some more interactions to the section navigation component (jumping to a specific section in addition to advancing one at a time). This PR was already getting big enough, though, so I will defer that stuff for the next one.

Also in this PR I removed all of the "data portal" stuff since that is now out of scope for the first release.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> part of #276 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: spotlight-client/src/SystemNarrativePage/SectionNavigation.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        <div>
+          <Chevron direction="up" faded />
+        </div>
+      ) : (
+        <Link
+          // the to props on these links is mainly for accessibility purposes, so they
+          // look like normal links. calling the setter is the critical step
+          to={`${urlBase}/${activeSection - 1}`}
+          onClick={() => {
+            setActiveSection(activeSection - 1);
```
Comment (daschi) [inline spotlight-client/src/SystemNarrativePage/SectionNavigation.tsx:144]: Do you need to add `e.preventDefault()` here too?

File: spotlight-client/src/SystemNarrativePage/SystemNarrativePage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    within(navRegion).getAllByText(`0${narrativeContent.sections.length + 1}`)
+      .length
+  ).toBe(2);
+
+  // no next on the last section
+  expect(nextLink).not.toBeInTheDocument();
+
+  // Jest/JSDOM don't support layout features
+  // so we can't really test anything related to scroll position :(
+  // (this also includes URL management because of how the page works)
```
Comment (daschi) [inline spotlight-client/src/SystemNarrativePage/SystemNarrativePage.test.tsx:87]: Does this mean you wouldn't be able to look at the `history` returned from `renderNavigableApp` to assert the page URL?

**Review by daschi — APPROVED** [review]
The transitions seem pretty smooth! It's great that the navigation for the sections is keyboard accessible. I just had a couple small questions.

File: spotlight-client/src/SystemNarrativePage/SectionNavigation.tsx (hunk 1 above)
Comment (macfarlandian) [reply spotlight-client/src/SystemNarrativePage/SectionNavigation.tsx:144]: no actually I should have removed it in both places! and the preventdefault was what was causing the URL history to mysteriously not update in reference to your second comment, which I spent like ....... an hour-plus trying to debug. wow

**Pull request merged**